### PR TITLE
[ENH] change path and name of subject level GLM folders

### PR DESCRIPTION
**Pull request by Remi-Gau**
Not ready yet...

---

Now it is good.

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Brief description.

**Normal case**

When the model name is the same as the task (the default) or when it is empty

```matlab
funcFWFM = 0;

opt.taskName = 'vislocalizer;'
opt.model.file: '/home/remi/github/CPP_SPM/tests/utils/../dummyData/models/model-vislocalizer_smdl.json'

ffxDir = getFFXdir(subLabel, funcFWFM, opt);

expectedOutput = 'task-vislocalizer_space-MNI_FWHM-0';
```

Model file
```json
{
  "Name": "vislocalizer",
  "Description": "contrasts for the visual localizer",
  "Input": {
    "task": "vislocalizer"
  }
}
```

**Comment by Remi-Gau** [comment]
When the model name does not match the talk name it is used as additional description (`desc`) of the GLM folder:

```matlab
funcFWFM = 6;

opt.taskName = 'nback;'
opt.model.file: '/home/remi/github/CPP_SPM/tests/utils/../dummyData/models/model-nback_smdl.json'

ffxDir = getFFXdir(subLabel, funcFWFM, opt);

expectedOutput = 'task-nback_space-individual_FWHM-6_desc-nbackMVPA';
```

Model file
```json
{
  "Name": "nback MVPA",
  "Description": "for folder naming",
  "Input": {
    "task": "nback"
  }
}
```

File: src/subject_level/getFFXdir.m [hunk 1]
```diff
@@ -18,14 +18,18 @@
   % :returns: - :ffxDir: (string)
   %
 
+  glmDirName = createGlmDirName(opt, funcFWFM);
+
+  model = spm_jsonread(opt.model.file);
+  if ~isempty(model.Name) && ~strcmpi(model.Name, opt.taskName)
+    glmDirName = [glmDirName, '_desc-', converToValidCamelCase(model.Name)];
+  end
```
Comment (CerenB) [inline src/subject_level/getFFXdir.m:26]: if I understood right, if `model.Name` exists in `model.json` file, and it is different then than `taskname`, the pipeline will create a name e.g. `RhythmBlock-MNI-FWHM-0-` and append `model.Name` ?

**Review by CerenB — APPROVED** [review]
with a short test run, it works in my dataset! 
it overwrites when the `model.Name` and `opt.taskname` are the same.

File: src/subject_level/getFFXdir.m (hunk 1 above)
Comment (Remi-Gau) [reply src/subject_level/getFFXdir.m:26]: yup exactly, except that it will append it `_desc-[model.Name]`

**Comment by Remi-Gau** [comment]
Cool will merge then

**Pull request merged**